pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -856,6 +879,56 @@ func (r *ReconcileClusterDeployment) setupClusterInstallServiceAccount(namespace
 	return currentSA, nil
 }
 
+func selectorPodWatchHandler(a handler.MapObject) []reconcile.Request {
+	retval := []reconcile.Request{}
+
+	pod := a.Object.(*corev1.Pod)
+	if pod == nil {
+		// Wasn't a Pod, bail out. This should not happen.
+		log.Errorf("Error converting MapObject.Object to Pod. Value: %+v", a.Object)
+		return retval
+	}
+	if pod.Labels != nil {
+		if cdName, ok := pod.Labels[install.ClusterDeploymentNameLabel]; ok {
+			retval = append(retval, reconcile.Request{NamespacedName: types.NamespacedName{
+				Name:      cdName,
+				Namespace: pod.Namespace,
+			}})
+		}
+	}
+	return retval
+}
+
+// Calculates the restart count for both hive and installer container for a cd and sets it to metricClusterDeploymentInstallRetries
+func (r *ReconcileClusterDeployment) updateInstallRetriesTotalMetric(clusterDeploymentName string, clusterDeploymentNamespace string, cdLog log.FieldLogger) {
+	labels := map[string]string{"cluster_deployment": clusterDeploymentName, "namespace": clusterDeploymentNamespace}
+	installerPodLabels := map[string]string{install.ClusterDeploymentNameLabel: clusterDeploymentName, install.InstallJobLabel: "true"}
+	_, err := metricClusterDeploymentInstallRetries.GetMetricWith(labels)
+	if err != nil {
+		cdLog.Error("Error getting cluster deployment install retries total metrics.")
+	} else {
+		var parsedLabels klabels.Selector
+		parsedLabels, err = klabels.Parse(klabels.Set(installerPodLabels).String())
+		pods := &corev1.PodList{}
+		err = r.Client.List(context.Background(), &client.ListOptions{Namespace: clusterDeploymentNamespace, LabelSelector: parsedLabels}, pods)
+		if err != nil {
+			log.WithError(err).Error("error listing pods")
+		} else {
+			podRestarts := 0
+			for _, pod := range pods.Items {
+				for _, cs := range pod.Status.ContainerStatuses {
+					podRestarts = podRestarts + int(cs.RestartCount)
+				}
+			}
+			if podRestarts > 0 {
+				cdLog.WithField("Installer Pod Restarts:", podRestarts).Warn("installer pod restarts > 1")
```

Comment:
Logger fields should be easy to search for in splunk or kibana, this will show up as "Installer Pod Restarts:=1", it should be more something like "restarts=1", but see the full example I sent as a comment in the PR.